### PR TITLE
WR-1279 add waiting room zone-level settings

### PR DESCRIPTION
**Pull request by piperswe**

### CODE REVIEW

**Comment by piperswe** [comment]
tested locally and working fine

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.20
 
 require (
 	github.com/agext/levenshtein v1.2.3 // indirect
-	github.com/cloudflare/cloudflare-go v0.66.0
+	github.com/cloudflare/cloudflare-go v0.66.1-0.20230508010110-9388783618dc
```
Comment (jacobbednarz) [inline go.mod:7]: please remove this from your PR. we handle these dependency changes outside of feature PRs to avoid conflicts.

File: internal/sdkv2provider/resource_cloudflare_waiting_room_settings.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		DeleteContext: schema.NoopContext,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceCloudflareWaitingRoomSettingsImport,
+		},
+
+		Schema: resourceCloudflareWaitingRoomSettingsSchema(),
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Second),
+			Update: schema.DefaultTimeout(30 * time.Second),
+		},
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_waiting_room_settings.go:28]: are these required? the default timeouts are currently 20 minutes.

File: internal/sdkv2provider/resource_cloudflare_waiting_room_settings.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	searchEngineCrawlerBypass := d.Get("search_engine_crawler_bypass").(bool)
+	return cloudflare.UpdateWaitingRoomSettingsParams{
+		SearchEngineCrawlerBypass: &searchEngineCrawlerBypass,
+	}
+}
+
+func resourceCloudflareWaitingRoomSettingsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Id()
+	if zoneID == "" {
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_waiting_room_settings.go:43]: this shouldn't ever happen; `d.Id()` should always be set in a `Read` operation.

File: internal/framework/service/rulesets/resource.go [hunk 4]
```diff
@@ -658,7 +658,7 @@ func toRulesetResourceModel(ctx context.Context, zoneID, accountID basetypes.Str
 			if ruleResponse.ActionParameters.FromValue != nil {
 				rule.ActionParameters[0].FromValue = []*ActionParameterFromValueModel{{
 					StatusCode:          flatteners.Int64(int64(ruleResponse.ActionParameters.FromValue.StatusCode)),
-					PreserveQueryString: flatteners.Bool(&ruleResponse.ActionParameters.FromValue.PreserveQueryString),
```
Comment (jacobbednarz) [inline internal/framework/service/rulesets/resource.go:661]: please remove these changes as they are unrelated to this PR.

File: go.mod (hunk 1 above)
Comment (piperswe) [reply go.mod:7]: this was temporary, was waiting on the cloudflare-go release. now that it's out I'll remove the change

**Comment by jacobbednarz** [comment]
acceptance tests all passing
```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareWaitingRoomSettings_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareWaitingRoomSettings_Create
=== PAUSE TestAccCloudflareWaitingRoomSettings_Create
=== CONT  TestAccCloudflareWaitingRoomSettings_Create
--- PASS: TestAccCloudflareWaitingRoomSettings_Create (15.69s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	16.720s
```

**Comment by jacobbednarz** [comment]
cheers @piperswe 🎈
